File: queue.py
import time 
import packet 
import threading
import globals
# import 

# lock = threading.Lock()



def enqueuePacket(pkt):
   # print len(globals.INGRESS_CAP)
   if ((globals.INGRESS_CAP[pkt.ingress].availableBuffSpace - pkt.packet_len) > 0):
      # lock[pkt.dst].acquire()
      globals.RECEIVE_COUNTER[pkt.ingress] +=1
      globals.INGRESS_CAP[pkt.ingress].availableBuffSpace -= pkt.packet_len
      
      # lock[pkt.dst].release()
   else:
      dropPacket(pkt)


def processPacket():
   for i in xrange(0,globals.INGRESS_LOC):
      if(globals.RECEIVE_COUNTER[i] > 0):
         globals.INGRESS_CAP[i].availableBuffSpace += globals.PKT_LEN*globals.INGRESS_CAP[i].numOfDequeuePkts
         if(globals.INGRESS_CAP[i].availableBuffSpace > globals.INGRESS_CAP[i].vmQueue):
            globals.INGRESS_CAP[i].availableBuffSpace = globals.INGRESS_CAP[i].vmQueue
      
      # constants.processCounter[pkt.dst] += INGRESS_CAP[pkt.dst].numOfDequeuePkts


def dropPacket(pkt):
   if (pkt.attack_flag == 0):
      globals.legitimateDropCounter[pkt.ingress]+=1
   else:
      globals.attackDropCounter[pkt.ingress] +=1

File: simulator(js)/queue.js
import time 
import packet 
import threading

lock = threading.Lock()


def receivepkt(pkt):
   if ((BUFFSIZE - pkt.packet_len) > 0):
      print_lock.acquire()
      receiveCounter+=1
      BUFFSIZE -= pkt.packet_len
      print_lock.release()
   else:
      droppkt()


def processedpkt(pkt,processingdelay):
   time.sleep(1/processingdelay)
   BUFFSIZE += pcket_len
   processCounter+=1

def droppkt(pkt):
   if (pkt.attack_flag == 0):
      legitimate_dropCounter+=1
   else:
      attack_dropCounter+=1

